kvrocks.conf

```diff
@@ -174,6 +174,20 @@ slave-read-only yes
 # By default the priority is 100.
 slave-priority 100
 
+# Change the default timeout in milliseconds for socket connect during fullsync.
+# The default value is 3100, and 0 means no timeout.
+#
+# If the master is unreachable before connecting, not having a timeout may block future
+# 'clusterx setnodes' commands because the replication thread is blocked on connect.
+slave-fullsync-connect-timeout 3100
+
+# Change the default timeout in milliseconds for socket recv during fullsync.
+# The default value is 3200, and 0 means no timeout.
+#
+# If the master is unreachable when fetching SST files, not having a timeout may block
+# future 'clusterx setnodes' commands because the replication thread is blocked on recv.
+slave-fullsync-recv-timeout 3200
```

Comment:
There also another socket connection in `ReplicationThread::CallbacksStateMachine`. Could you unify these connection timeout settings in replication and name this config option `replication-..-timeout`? cc @git-hulk

Reply:
For connection timeout I think it's good to share the config. Not sure if it's fine for read/write timeout. WDYT?

Reply:
The main replication thread is also using a blocking socket for receiving data, so it may block if the master is unreachable. I will test this case and modify it if the situation exists.

Reply:
> There's also another socket connection in ReplicationThread::CallbacksStateMachine. Could we unify these connection timeout settings in replication and name this config option replication-..-timeout? cc @git-hulk

Yes, it would be better to use a unify connection timeout for replication sockets.

Reply:
The main replication thread seems to be using a blocking socket wrapped by bufferevent for receiving data, so it may block if master is unreachable. I will test this case and modify it if the situation exists.

Reply:
The replication socket is a blocking socket, but it won't block forever if we have set the timeout when connecting.

Reply:
I unified the connection timeout config. Please have a look.